peer.go

```diff
@@ -205,16 +228,22 @@ func (p *Peer) connect() error {
 		p.readConn = conn
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	p.cancelWriteHandler = cancel
+	ctx := context.Background()
+
+	writerCtx, cancelWriter := context.WithCancel(ctx)
```

Comment:
@boecklim Shouldn't you be passing in the global context here? Even if you are creating a new one, this one should end when the global context is Done.

Reply:
@icellan I think, I know what you mean. Added another commit. Please take a look.